src/main/java/bio/terra/service/dataset/flight/ingest/IngestParseJsonFileStep.java

```diff
@@ -0,0 +1,101 @@
+package bio.terra.service.dataset.flight.ingest;
+
+import bio.terra.common.Column;
+import bio.terra.model.BulkLoadFileModel;
+import bio.terra.model.IngestRequestModel;
+import bio.terra.model.TableDataType;
+import bio.terra.service.dataset.Dataset;
+import bio.terra.service.dataset.exception.IngestFailureException;
+import bio.terra.service.filedata.google.gcs.GcsPdao;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import bio.terra.stairway.exception.RetryException;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class IngestParseJsonFileStep implements Step {
+
+  private final GcsPdao gcsPdao;
+  private final Dataset dataset;
+  private final ObjectMapper objectMapper;
+
+  public IngestParseJsonFileStep(GcsPdao gcsPdao, ObjectMapper objectMapper, Dataset dataset) {
+    this.gcsPdao = gcsPdao;
+    this.objectMapper = objectMapper;
+    this.dataset = dataset;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext flightContext)
+      throws InterruptedException, RetryException {
+    IngestRequestModel ingestRequest = IngestUtils.getIngestRequestModel(flightContext);
+    List<String> gcsFileLines =
+        gcsPdao.getGcsFilesLines(
```

Comment:
There are probably many cases like this in TDR, but it looks like we are reading a file completely into memory without checking its size first. If so, a very large file could crash TDR.

Reply:
Adding a simple check here would be a quick way to warn users against using this methodology for their larger ingests. They should use the separate steps for large ingests.

Reply:
(check is added now!)